### ISSUE TITLE
Database not created due to the port error 

### ISSUE THREAD

**Issue by creeper00** [creeper00/Upenl_newb#197]
<a href="https://github.com/KRKimDongWoo"><img src="https://avatars.githubusercontent.com/u/20403194?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [KRKimDongWoo](https://github.com/KRKimDongWoo)**
_Sunday Feb 02, 2020 at 12:42 GMT_
_Originally opened as https://github.com/snuspl/crane/issues/27_

----

## Environment
- Ubuntu18.04 LTS
- python3.7, with all dependencies in setup.py

## Version
- origin/master branch

### cluster_info.yaml
``` 
master:
    url: "127.0.0.1:55555"

nodes:
    - url: "127.0.0.1"
      resource:
          gpus: 4
```
### Command
`
$ crane.cluster role=setup cluster_info=./cluster_info.yaml pid.dir=/tmp/crane run.debug=true
`

### crane-node.log
```
[2020-02-02 21:33:46,735][crane.node.run][INFO] - : 
master:
  url: 127.0.0.1:55555
name: crane-node
pid:
  dir: /tmp/crane
resource:
  gpus: 4
  rdma: false
role: setup
run:
  debug: true
  detach: true
server:
  url: 127.0.0.1
swarm:
  manager: 127.0.0.1
  token: SWMTKN-1-4zcr1bfmivutdmy9ivq5ag27wtupgts7h0vyh25bgo6oxl1faj-47g4y962zngp9hivup8twcbfy

[2020-02-02 21:33:46,735][crane.node.run][INFO] - Working directory: /home/dongwoo/.crane/outputs/2020-02-02/21-33-46
[2020-02-02 21:33:47,779][crane.node.run][ERROR] - IndexError('list index out of range')
[2020-02-02 21:33:47,799][crane.node.run][ERROR] - Traceback (most recent call last):
  File "/home/dongwoo/development/crane/crane/crane/node/run.py", line 121, in main
    asyncio.run(script)
  File "/usr/lib64/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "uvloop/loop.pyx", line 1456, in uvloop.loop.Loop.run_until_complete
  File "/home/dongwoo/development/crane/crane/crane/node/run.py", line 158, in run_setup
    log_ep = await _run_log_service(node_endpoint)
  File "/home/dongwoo/development/crane/crane/crane/node/run.py", line 264, in _run_log_service
    log_ep, _ = await run_and_get_endpoint(name, option)
  File "/home/dongwoo/development/crane/crane/crane/docker.py", line 137, in run_and_get_endpoint
    port_info = list(map(lambda x: x[0], filter(bool, port_mapping_info.values())))[0]
IndexError: list index out of range
```
### crane-manager.log
```
[2020-02-02 21:33:17,597][crane.master.run][INFO] - : 
db:
  url: mysql://CRANE_DB_USER:CRANE_DB_PASSWORD@127.0.0.1:42117/CRANE_DB
executor:
  class: crane.master.executor.distributed.DistributedExecutor
log:
  port: null
master:
  url: 127.0.0.1:55555
name: crane-manager
pid:
  dir: /tmp/crane
profile:
  port: null
role: core
run:
  debug: true
  detach: true

[2020-02-02 21:33:17,598][crane.master.run][INFO] - Working directory: /home/dongwoo/.crane/outputs/2020-02-02/21-33-17
[2020-02-02 21:33:17,607][root][INFO] - Master configuration: Config(master=EndPoint(ip='127.0.0.1', port=55555), db_url='mysql://CRANE_DB_USER:CRANE_DB_PASSWORD@127.0.0.1:42117/CRANE_DB', executor='DistributedExecutor')
[2020-02-02 21:33:17,608][tortoise][INFO] - Tortoise-ORM startup
    connections: {'default': {'engine': 'tortoise.backends.mysql', 'credentials': {'port': 42117, 'charset': 'utf8mb4', 'sql_mode': 'STRICT_TRANS_TABLES', 'database': 'CRANE_DB', 'host': '127.0.0.1', 'user': 'CRANE_DB_USER', 'password': 'CRANE_DB_PASSWORD'}}}
    apps: {'models': {'models': ['crane.db'], 'default_connection': 'default'}}
[2020-02-02 21:33:17,634][crane.master.core][INFO] - Starting crane master core.
[2020-02-02 21:33:17,635][crane.master.core][INFO] - Scheduler cancelled, running cleanup.
[2020-02-02 21:33:17,635][crane.master.core][INFO] - Scheduler exited.
```
### crane-cluster.log
```
[2020-02-02 21:34:02,308][crane.cluster.run][INFO] - : 
all: false
name: crane-cluster
nodes: 127.0.0.1
pid:
  dir: /tmp/crane
role: teardown
run:
  debug: false

[2020-02-02 21:34:02,309][crane.util.process][INFO] - $ ssh -p 22 127.0.0.1 exit
[2020-02-02 21:34:02,314][crane.cluster.run][ERROR] - FileNotFoundError(2, 'No such file or directory')
[2020-02-02 21:34:02,315][crane.cluster.run][ERROR] - Traceback (most recent call last):
  File "/home/dongwoo/development/crane/crane/crane/cluster/run.py", line 80, in main
    asyncio.run(script)
  File "/usr/lib64/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "uvloop/loop.pyx", line 1456, in uvloop.loop.Loop.run_until_complete
  File "/home/dongwoo/development/crane/crane/crane/cluster/run.py", line 159, in teardown
    with open(pidfile, "r") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/crane/crane-manager.pid'
```

**Comment by creeper00** [comment]
<a href="https://github.com/yuyupopo"><img src="https://avatars.githubusercontent.com/u/9143939?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [yuyupopo](https://github.com/yuyupopo)**
_Monday Feb 03, 2020 at 06:27 GMT_

----

I can reproduce this error when using a hotspot network. Was this your case?

**Comment by creeper00** [comment]
<a href="https://github.com/KRKimDongWoo"><img src="https://avatars.githubusercontent.com/u/20403194?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [KRKimDongWoo](https://github.com/KRKimDongWoo)**
_Monday Feb 03, 2020 at 12:27 GMT_

----

@yuyupopo  This error happened both in Wi-fi and LAN network

**Comment by creeper00** [comment]
<a href="https://github.com/yuyupopo"><img src="https://avatars.githubusercontent.com/u/9143939?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [yuyupopo](https://github.com/yuyupopo)**
_Monday Feb 03, 2020 at 23:49 GMT_

----

This is a very strange error as the docker container cannot bind the requested address. 
I've seen that it also results in node manager failure, as it cannot bind the address also.

**Comment by creeper00** [comment]
<a href="https://github.com/yuyupopo"><img src="https://avatars.githubusercontent.com/u/9143939?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [yuyupopo](https://github.com/yuyupopo)**
_Tuesday Feb 11, 2020 at 13:09 GMT_

----

I think this issue is resolved with #41 .
I'll close this issue, reopen it if same issue occurs again.
